bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,325 @@
+        # command to add spookified name
+        help_embed.add_field(
+            name="How do I add my spookified word?",
+            value=f"Simply type `{self.bot.command_prefix}spooknamerate add my word`",
+            inline=False
+        )
+
+        help_embed.add_field(
+            name="How do I *delete* my spookified word?",
+            value=f"Simply type `{self.bot.command_prefix}spooknamerate delete`",
+            inline=False
+        )
+
+        await ctx.send(embed=help_embed)
+
+    @spook_name_rate.command(name='list', aliases=["all", "entries"])
+    async def list_entries(self, ctx: Context) -> None:
+        """Send all the entries up till now in a single embed."""
+        await ctx.send(embed=await self.get_responses_list(final=False))  # Send a list of users
+
+    @spook_name_rate.command(name='word')
+    async def tell_word(self, ctx: Context) -> None:
+        """Tell the current random word (name)."""
+        if not self.poll:
+            await ctx.send(f"The name is **{self.word}**")
+            return
+        else:
+            await ctx.send(
+                f"The word ~~is~~ was **{self.word}**. The poll has already started, so you cannot "
+                "add an entry."
+            )
+            return
+
+    @spook_name_rate.command(name="add", aliases=["+", "register"])
+    async def add_word(self, ctx: Context, *, word: str) -> None:
+        """A command that adds your word!"""
+        if self.poll:
+            logger.info(f"{ctx.message.author} tried to add a word, but the poll had already started.")
+            await ctx.send("Sorry, the poll has started! You can try and participate in the next round though!")
+            return
+
+        message = ctx.message  # get the message
+
+        for message_id in self.messages:
+            data = self.messages.get(message_id)
+            if data['author'] == message.author:  # if the author has already added an entry
+                await ctx.send(
+                    "But you have already added an entry! Type "
+                    f"`{self.bot.command_prefix}spooknamerate "
+                    "delete` to delete it, and then you can add it again"
+                )
+                return
+
+            elif data['word'] == word:
+                # Someone has added the word
+                await ctx.send("TOO LATE. Someone has already added this word.")
+                return
+
+        # otherwise
+        self.messages[message.id] = {  # store it
+            'word': word,
+            'author': message.author,
+            'score': 0,
+        }
+
+        for emoji in self.emojis_val:
+            await message.add_reaction(emoji)  # add each emoji
+
+        logger.info(f"{message.author} added the word {word!r}")
+
+        await ctx.send(f"{word!r} added successfully!\n{random.choice(self.added_messages)}")
+
+    @Cog.listener()
+    async def on_reaction_add(self, reaction: discord.Reaction, user: discord.User) -> None:
```

Comment:
```suggestion
    async def on_reaction_add(self, reaction: Reaction, user: User) -> None:
```
I do believe we should be consistent if we use `discord.Something` or just `Something`, given you're using typings like `TextChannel`, `Context` and `Message` already, I would like to see these without discord. (and probably other places)